packages/proxy/src/util/responseHelpers.ts

```diff
@@ -0,0 +1,46 @@
+import type { ServerResponse } from 'node:http';
+import { DiscordAPIError, parseResponse, RateLimitError } from '@discordjs/rest';
+import type { Dispatcher } from 'undici';
+
+/**
+ * Populates a server response with the data from a Discord 2xx REST response
+ * @param res The server response to populate
+ * @param data The data to populate the response with
+ */
+export async function populateOkResponse(res: ServerResponse, data: Dispatcher.ResponseData): Promise<void> {
+	res.statusCode = data.statusCode;
+
+	for (const header of Object.keys(data.headers)) {
```

Comment:
Use Object.entries

Reply:
If we always used the value, sure, but we're skipping them. As such, for performance reasons, it may be wise to keep this as is.

Reply:
Is there any actual performance hit tho?

Reply:
Considering it has to allocate an array for every property... most likely?